### ISSUE TITLE
Fax issue in 0.1.21: `System.ArgumentException: boundary`

### ISSUE THREAD

**Issue by grokify** [ringcentral/ringcentral-csharp#36]
Version 0.1.21 is currently returning the following error when `GetHttpContent()` is called in the `Request` object.

Please also add mock & real tests for sending faxes in the test suite.

```cs
var request = new Request ("/restapi/v1.0/account/~/extension/~/fax", json, attachments);
var body = request.GetHttpContent ()
```

Exception:

```
Unhandled Exception:
System.ArgumentException: boundary
  at System.Net.Http.MultipartContent..ctor (System.String subtype, System.String boundary) <0x4561278 + 0x00257> in <filename unknown>:0 
  at System.Net.Http.MultipartFormDataContent..ctor (System.String boundary) <0x4561248 + 0x0001f> in <filename unknown>:0 
  at RingCentral.SDK.Http.Request.GetHttpContent () <0x7a7a88 + 0x00187> in <filename unknown>:0 
  at faxCustomCoverPage.MainClass.SendFax (RingCentral.SDK.SDK sdk) <0x4560788 + 0x00153> in <filename unknown>:0 
  at faxCustomCoverPage.MainClass.Main (System.String[] args) <0x559ec8 + 0x00013> in <filename unknown>:0 
[ERROR] FATAL UNHANDLED EXCEPTION: System.ArgumentException: boundary
  at System.Net.Http.MultipartContent..ctor (System.String subtype, System.String boundary) <0x4561278 + 0x00257> in <filename unknown>:0 
  at System.Net.Http.MultipartFormDataContent..ctor (System.String boundary) <0x4561248 + 0x0001f> in <filename unknown>:0 
  at RingCentral.SDK.Http.Request.GetHttpContent () <0x7a7a88 + 0x00187> in <filename unknown>:0 
  at faxCustomCoverPage.MainClass.SendFax (RingCentral.SDK.SDK sdk) <0x4560788 + 0x00153> in <filename unknown>:0 
  at faxCustomCoverPage.MainClass.Main (System.String[] args) <0x559ec8 + 0x00013> in <filename unknown>:0
```

Here are my files for reference:

1. [Test code](https://github.com/grokify/ringcentral-demos-fax-cover-page/blob/master/csharp/csharp/Program.cs)
2. [Test repo](https://github.com/grokify/ringcentral-demos-fax-cover-page/)

**Comment by tylerlong** [comment]
The following works for me with `1.0.0` branch:

```cs
[Test]
public void Fax()
{
    var requestBody = JsonConvert.SerializeObject(new
    {
        faxResolution = "High",
        coverIndex = 0,
        to = new object[] { new { phoneNumber = Config.Instance.Receiver } }
    });

    var attachments = new List<Attachment>();
    var textBytes = System.Text.Encoding.UTF8.GetBytes("hello fax");
    var attachment = new Attachment(@"test.txt", "application/octet-stream", textBytes);
    attachments.Add(attachment);
    var request = new Request("/restapi/v1.0/account/~/extension/~/fax", requestBody, attachments);
    var body = request.HttpContent;
    Assert.NotNull(body);
    var response = sdk.Platform.Post(request);
    Assert.AreEqual(true, response.OK);
}
```

**Comment by tylerlong** [comment]
The csharp demo runs pretty well on Windows. But an exception was thrown when running on Mac:

```
System.ArgumentException: boundary.
```

Inspired by this [Stackoverflow thread](http://stackoverflow.com/questions/29075414/system-argumentexception-when-uploading-multipart). I finally figured out the root cause and fixed it:  https://github.com/ringcentral/ringcentral-csharp/commit/3cceffcaa4929fd8f3976b3ec0115edd96189514

The root cause is Mono on Mac doesn't like the hard-coded boundary string: `Boundary_1_14413901_1361871080888 `. It thinks that this boundary string is invalid so it throws `System.ArgumentException`.  I fixed it by removing the hard-coded boundary string. Let it generate a dynamic boundary string is way better.

Let's release C# SDK 1.0.0-alpha2 as soon as possible. Currently I host it locally.

**Comment by tylerlong** [comment]
By the way, I don't think the hard-coded boundary string `Boundary_1_14413901_1361871080888 ` is invalid. And that's why the app runs without problem on Windows.

But the Mono implementation of the `MultipartFormDataContent` class doesn't like it for some reason.

**Comment by grokify** [comment]
Closing due ot verified fix in 1.0.0-alpha2.
